### PR TITLE
Jest integration

### PR DESCRIPTION
**Pull request by NimaSoroush**
#38 
- Add jest snapshot support
- Performance improvement
- Better logging

### CODE REVIEW

File: src/compareImage.js [hunk 1]
```diff
@@ -1,53 +1,82 @@
...
+  }
+  prefixedLogger.log(`screenshot saved in -> ${snapshotPath}`);
+  if (fs.existsSync(diffPath)) {
+    fs.unlinkSync(diffPath);
+  }
+  if (!fs.existsSync(snapshotsDir)) {
+    fs.mkdirSync(snapshotsDir);
+  }
+  fs.writeFileSync(snapshotPath, capturedImage);
+  result = testConfig.isUpdate ? { updated: true } : { added: true };
```
Comment (danrr) [inline src/compareImage.js]: this `result` variable is declared at the top but then overwritten here.

Declare it here or remove it and just return the value

Comment (NimaSoroush) [reply src/compareImage.js]: Good point. That was left over from my refactoring. will fix it

File: src/page.js [hunk 2]
```diff
@@ -120,12 +110,35 @@ export default class Page {
...
   async _compareImage() {
     if (!this.error) {
+      let result;
       try {
-        return await compareImage(this.globalConfig, this.config.testName);
+        result = await compareImage(this.image, this.globalConfig, this.testConfig);
       } catch (error) {
         this._logError(error);
+        throw error;
+        // return false;
```
Comment (danrr) [inline src/page.js]: why is this commented out?

Comment (NimaSoroush) [reply src/page.js]: lol, left over again. `throw error;` should go away as well

File: src/scripts/checkVersion.js [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+    console.log(`Please follow Differencify's contributing guide (see ${contributingLink}).`);
+    console.log('');
+    process.exit(1);
+  }
+};
+
+console.log('Checking Node & npm versions...');
+console.log('');
+
+checkVersion('node --version', 'Node', '8.6.0');
```
Comment (danrr) [inline src/scripts/checkVersion.js]: does this need to be this strict?

Comment (NimaSoroush) [reply src/scripts/checkVersion.js]: This is for development of this package. I want to make sure people run and test with same page and not getting different results. This been added to .nvmrc too

Comment (NimaSoroush) [reply src/scripts/checkVersion.js]: I've added a note to contributing.md as well

Comment (danrr) [reply src/scripts/checkVersion.js]: yeah, it's fine, I just discovered you can do `nvm use` with nvmrc files.

File: src/page.js (hunk 2 above)
Comment (NimaSoroush) [reply src/page.js]: Done

File: src/compareImage.js (hunk 1 above)
Comment (NimaSoroush) [reply src/compareImage.js]: Done

**Pull request merged**